### DISCUSSION TITLE
After completing a swap transaction, how do I update the liquidity amount L?

### DISCUSSION THREAD

**Discussion by vball** [Jeiwan/uniswapv3-book#86] [Milestone 1. First Swap]
Do I need to recalculate the liquidity amount L and record the reduced portion?

**Comment by OoXooOx** [comment]
Yeah, it's not clear from 1 milestone, but you can check code of milestone_6.
In swap function in UniswapV3Pool you can see how we work with liquidity. 
uint128 liquidity_ = liquidity;
if (zeroForOne) liquidityDelta = -liquidityDelta;
state.liquidity = LiquidityMath.addLiquidity(
                    state.liquidity,
                    liquidityDelta
                );
  if (liquidity_ != state.liquidity) liquidity = state.liquidity;
